[Bug]: SetState fails on FlinkRunner

### What happened?

```
Caused by: java.lang.ClassCastException: class java.lang.Boolean cannot be cast to class java.lang.Void (java.lang.Boolean and java.lang.Void are in module java.base of loader 'bootst
rap')
        at org.apache.flink.api.common.typeutils.base.VoidSerializer.serialize(VoidSerializer.java:30)
        at org.apache.flink.contrib.streaming.state.AbstractRocksDBState.serializeValueInternal(AbstractRocksDBState.java:158)
        at org.apache.flink.contrib.streaming.state.AbstractRocksDBState.serializeValueNullSensitive(AbstractRocksDBState.java:175)
        at org.apache.flink.contrib.streaming.state.RocksDBMapState.put(RocksDBMapState.java:136)
        at org.apache.beam.runners.flink.translation.wrappers.streaming.state.FlinkStateInternals$FlinkSetState.add(FlinkStateInternals.java:1468)
```

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [X] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/state/FlinkStateInternals.java
+++ b/runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/state/FlinkStateInternals.java
@@ -76,7 +76,6 @@ import org.apache.flink.api.common.state.ValueStateDescriptor;
 import org.apache.flink.api.common.typeutils.TypeSerializer;
 import org.apache.flink.api.common.typeutils.base.BooleanSerializer;
 import org.apache.flink.api.common.typeutils.base.StringSerializer;
-import org.apache.flink.api.common.typeutils.base.VoidSerializer;
 import org.apache.flink.runtime.state.KeyedStateBackend;
 import org.apache.flink.runtime.state.VoidNamespace;
 import org.apache.flink.runtime.state.VoidNamespaceSerializer;
@@ -1409,7 +1408,9 @@ public class FlinkStateInternals<K> implements StateInternals {
       this.flinkStateBackend = flinkStateBackend;
       this.flinkStateDescriptor =
           new MapStateDescriptor<>(
-              stateId, new CoderTypeSerializer<>(coder, pipelineOptions), new BooleanSerializer());
+              stateId,
+              new CoderTypeSerializer<>(coder, pipelineOptions),
+              BooleanSerializer.INSTANCE);
     }
 
     @Override
@@ -1612,7 +1613,7 @@ public class FlinkStateInternals<K> implements StateInternals {
             new MapStateDescriptor<>(
                 id,
                 new CoderTypeSerializer<>(elemCoder, pipelineOptions),
-                VoidSerializer.INSTANCE));
+                BooleanSerializer.INSTANCE));
       } catch (Exception e) {
         throw new RuntimeException(e);
       }

--- a/website/www/site/layouts/shortcodes/flink_java_pipeline_options.html
+++ b/website/www/site/layouts/shortcodes/flink_java_pipeline_options.html
@@ -133,6 +133,11 @@ Should be called before running the tests.
   <td>Default: <code>false</code></td>
 </tr>
 <tr>
+  <td><code>operatorChaining</code></td>
+  <td>Sets the behavior of operator chaining.</td>
+  <td>Default: <code>true</code></td>
+</tr>
+<tr>
   <td><code>parallelism</code></td>
   <td>The degree of parallelism to be used when distributing operations onto workers. If the parallelism is not set, the configured Flink default is used, or 1 if none can be found.</td>
   <td>Default: <code>-1</code></td>

--- a/website/www/site/layouts/shortcodes/flink_python_pipeline_options.html
+++ b/website/www/site/layouts/shortcodes/flink_python_pipeline_options.html
@@ -133,6 +133,11 @@ Should be called before running the tests.
   <td>Default: <code>false</code></td>
 </tr>
 <tr>
+  <td><code>operator_chaining</code></td>
+  <td>Sets the behavior of operator chaining.</td>
+  <td>Default: <code>true</code></td>
+</tr>
+<tr>
   <td><code>parallelism</code></td>
   <td>The degree of parallelism to be used when distributing operations onto workers. If the parallelism is not set, the configured Flink default is used, or 1 if none can be found.</td>
   <td>Default: <code>-1</code></td>

